src/client/datascience/jupyter/kernels/kernelExecution.ts

```diff
@@ -87,6 +87,35 @@ export class KernelExecution implements IDisposable {
             traceInfo('Restore notebook state to idle after completion');
         }
     }
+
+    /**
+     * This is a temporary method to support interrupting kernels when code cells are executed
+     * outside of the kernelExecution/cellExecution codepath. Do not use!
+     */
+    public async interruptInteractiveKernel(document: NotebookDocument, notebookPromise?: Promise<INotebook>): Promise<InterruptResult> {
```

Comment:
What's the difference between this method and interrupt()? Seems almost the same.

Reply:
It creates a deferred if there's no execution queue and lets the execution time out.